File: pythonCode/bilateralFIlter.py
import cv2
import sys
import numpy as np

#read image
image = cv2.imread("../assets/putin_noise.jpg")

#check if image exits
if image is None:
    print("can not find image")
    sys.exit()

#define diameter of the pixel neighbourhood to be used for filter
diameter = 15

sigmaColor = 80

sigmaSpace = 80

#Apply bilateral filter
result = cv2.bilateralFilter(image, diameter, sigmaColor, sigmaSpace)

#create windows to display images
cv2.namedWindow("image", cv2.WINDOW_NORMAL)
cv2.namedWindow("result", cv2.WINDOW_NORMAL)

#display image
cv2.imshow("image", image)
cv2.imshow("result", result)

#press esc to exit the program
cv2.waitKey(0)

#close all the opened windows
cv2.destroyAllWindows()

File: cPlusPlusCode/bilateralFIlter.cpp
#include<opencv2/imgproc.hpp>
#include<opencv2/highgui.hpp>
#include<iostream>

using namespace std;
using namespace cv;

int main(){
    //Read image
    Mat image = imread("../assets/putin_noise.jpg");

    //check if image exists
    if(image.empty()){
        cout<<"can not find image"<<endl;
        return 0;
    }

    //define diameter of the pixel neighbourhood to be used for filter
    int diameter = 15;

    double sigmaColor = 80;

    double sigmaSpace = 80;

    Mat result;

    //Apply bilateral filter
    bilateralFilter(image, result, diameter, sigmaColor, sigmaSpace);

    //create windows to display images
    namedWindow("image", WINDOW_NORMAL);
    namedWindow("result", WINDOW_NORMAL);

    //display images
    imshow("image", image);
    imshow("result", result);

    //press esc to exit the program
    waitKey(0);

    //close all the opened windows
    destroyAllWindows();

    return 0;
}